testing/test_kerns.py

```diff
@@ -221,29 +221,38 @@ class TestSlice(unittest.TestCase):
     """
 
     def setUp(self):
-        tf.reset_default_graph()
         self.rng = np.random.RandomState(0)
-        self.k1 = GPflow.kernels.RBF(1, active_dims=[0])
-        self.k2 = GPflow.kernels.RBF(1, active_dims=[1])
-        self.k3 = GPflow.kernels.RBF(1)
+        tf.reset_default_graph()
+
         self.X = self.rng.randn(20, 2)
         self.Z = self.rng.randn(10, 2)
 
+        kernels = GPflow.kernels.Stationary.__subclasses__() + [GPflow.kernels.Constant, GPflow.kernels.Linear,
+                                                                GPflow.kernels.Polynomial]
+        self.kernels = []
+        for kernclass in kernels:
+            k1 = kernclass(self.X.shape[1], active_dims=[0])
```

Comment:
Ok, this is what I was really confused about, but I think it's consistent in my head now. So `input_dim` is the number of total active_dims, rather than the number of columns in `X`?